frameworks/kafka/docs/upgrade.md

```diff
@@ -0,0 +1,111 @@
+---
+post_title: Upgrade Kafka from 0.10.2.1 to 0.11.0.0
+menu_order: 10
+feature_maturity: preview
+enterprise: 'no'
+---
+
+# Upgrade Kafka from 0.10.2.1 to 0.11.0.0
+
+
+This document explains how to upgrade the DC/OS Kafka service from 0.10.2.x to 0.11.0.0 version. Refer to the Kafka [upgrade](https://kafka.apache.org/documentation/#upgrade) documentation for more information.
+
+- Information about the new message format in 0.11.0.0 is available [here](https://kafka.apache.org/documentation/#upgrade_11_message_format).
+
+- Kafka 0.11.0.0 introduces idempotent and transactional capabilities. These new 0.11.0.0 features require the new messaging format, and so will not work on older formats. 
+
+- 0.11.0.0 clients can communicate with older brokers, such as 0.10.2.x brokers. However 0.11.0.0 clients are required for these new features. 
+
+The rolling [upgrade procedure](https://kafka.apache.org/documentation/#upgrade) (from 0.10.2.x to 0.11.0.0) is explained below. This procedure guarantees no downtime during the upgrade.
+
+
+1. Upgrade the code to the new version:
+   * Upgrade Kafka software to [0.11.0.0](https://archive.apache.org/dist/kafka/0.11.0.0/RELEASE_NOTES.html).
+   * Restart the brokers one at a time.    
+   * Wait until the entire cluster is upgraded.
+
+2. Change `inter.broker.protocol.version` to `0.11.0.0`:
+
+   * In the `server.properties` file, change `inter.broker.protocol.version` to `0.11.0.0`.
+   * Do not change log.message.format.version yet.
+   * Restart the brokers one by one for the new protocol version to take affect.
+
+3. Change `log.message.format.version` to `0.11.0`:
+
+   * In the server.properties files, change `log.message.format.version` to `0.11.0`.
+   * Restart the brokers one by one for the new log format version to take affect.
+
+
+
+The new DC/OS Kafka package (with Kafka 0.11.0.0 version) that we will be upgrading to has the following default settings:
+
+* Kafka Version = 0.11.0.0
+* inter.broker.protocol.version = 0.11.0.0
+* log.message.format.version = 0.11.0
+
+The current DC/OS Kafka service (installed with Kafka 0.10.2.x version) will typically have the following configuration:
```

Comment:
Let's not call it `current` because the current package is the latest. Maybe use `earlier` or `previous`.

Let's not say "typically". Instead just state what the default values were in the previous release.